app/components/Reactions/ReactionPicker.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+
+// $FlowFixMe
+import React, { useMemo, useState, useCallback } from 'react';
+import type { EmojiEntity } from 'app/reducers/emojis';
+import type { ID } from 'app/models';
+import styles from './ReactionPicker.css';
+import ReactionPickerHeader from './ReactionPickerHeader';
+import ReactionPickerContent from './ReactionPickerContent';
+import ReactionPickerFooter from './ReactionPickerFooter';
+import emojiLoading from 'app/assets/emoji_loading.svg';
+
+type Props = {
+  isLoading: boolean,
+  emojis: Array<EmojiEntity>,
+  addReaction: ({
+    emoji: string,
+    contentTarget: string
+  }) => Promise<*>,
+  deleteReaction: ({ reactionId: ID, contentTarget: string }) => Promise<*>,
+  contentTarget: string
+};
+
+const ReactionPicker = ({
+  isLoading,
+  emojis,
+  addReaction,
+  deleteReaction,
+  contentTarget
+}: Props) => {
+  const [activeCategory, setActiveCategory] = useState(null);
+  const [searchString, setSearchString] = useState(null);
+
+  const categories = useMemo(() => {
+    if (!emojis) {
+      return {};
+    }
+    const mappedCategories = {};
+    let activeCategorySet = false;
+    emojis.forEach(emoji => {
+      if (!activeCategorySet) {
+        setActiveCategory(emoji.category);
+        activeCategorySet = true;
+      }
+      if (emoji.category in mappedCategories) {
+        mappedCategories[emoji.category]['emojis'].push(emoji);
+      } else {
+        mappedCategories[emoji.category] = {
+          name: emoji.category,
+          emojis: [emoji]
+        };
+      }
+    });
+    return mappedCategories;
+  }, [emojis]);
+
+  const searchResults = useMemo(() => {
+    if (searchString === null || searchString === '') {
+      return null;
+    }
+    return emojis.filter(emoji => {
+      const idMatches = emoji.shortCode.toLowerCase().includes(searchString);
```

Comment:
![fuzzy](https://user-images.githubusercontent.com/23152018/64199621-0586b100-ce8b-11e9-9730-af61a5a363e5.png)

Reply:
I've been thinking a bit about this an believe I have a proper solution:

`searchString`: `cat`

1. Look up exact matches for the `shortCode`:
  - `:cat:`
2. Look up word matches (i.e. look up if the `shortCode` starts/ends with the searchString and such) :
  - `:cat_crying:`
  - `:mad_cat:`
  - `:hungry_cat_sad:`
  - `:cat2:`
3. Perform a fuzzy search for the `shortCode`:
  - `:capitcal:`
  - `:cactus:`
  - `:bar_char:`

Append each result to a list, and pop the results from the original emoji list for each result.

I have not decided what to do with `keywords` yet, maybe just look them up if there are exact matches for the keywords, and insert this a separate step before the fuzzy search.